arch/arm64/src/common/arm64_arch.h

```diff
@@ -466,7 +465,7 @@ static inline int arch_get_exception_depth(void)
 
 static inline uint64_t arch_get_current_tcb(void)
 {
-  return read_sysreg(tpidr_el1);
+  return read_sysreg(tpidr_el0);
```

Comment:
You cannot rely on this as the user has access to tpidr_el0 and can corrupt the contents at any point. This can be considered a security risk as well.

Reply:
kernel space use tpidr_el1 here:
https://github.com/apache/nuttx/pull/13486/files#diff-a696b8576fe26629d886f8b4d2d3224143dc83018435ec1787300b2ba9fa6b73R404
tpidr_el0 will be used as userspace tls later.

Reply:
Ok, will you handle arch_get_current_tcb differently then? Using tpidr_el0 inside the kernel is dangerous as tpidr_el0 is in fact writable from el0 / user-space

Reply:
@hujun260 it's better to remove arch_get_current_tcb in this patch and bring back later when we really use it.

Reply:
ok